### PR TITLE
Various fixes for recent transaction proving

### PR DESCRIPTION
**Pull request by sisou**
## What's in this pull request?

Together with @jsdanielh we finally figured out why fetching recent transactions doesn't work reliably. It's again because of a mismatch of current head block between the browser client and the history nodes it requests from.

When we changed which block the full node is using to answer a tx-proof request from its current head to the block requested by the client in 058a9ec09e815d720fb201fa892ffa5374d0a538, we failed to see that this also affects micro blocks in the current batch that are older than the node's head block.

Closes #1515.

## Pull request checklist

- [ ] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: consensus/src/messages/handlers.rs [hunk 1]
```diff
@@ -465,7 +463,7 @@ fn prove_transaction(
 
         let block = blockchain
             .chain_store
-            .get_block_at(election_head, false, None)
+            .get_block_at(proving_block_number, false, None)
```
Comment (sisou) [inline consensus/src/messages/handlers.rs:466]: This one is also important, but I just found it as a drive-by. It's was probably a copy-paste oversight.

Comment (viquezclaudio) [reply consensus/src/messages/handlers.rs:466]: Yes indeed, we need to use `proving_block_number`

**Review by viquezclaudio — APPROVED** [review]
Nice

**Comment by viquezclaudio** [comment]
Do we still want to close #1515 ? 
I don't know if for now it is better to just log the warning without closing the connection, and leave the handling of malicious behaviour as part of hardening?
